test/distributed/_tensor/test_dtensor.py

```diff
@@ -420,27 +420,68 @@ def test_dtensor_spec_local_shard_offset(self):
 
     @with_comms
     def test_from_local_sub_mesh(self):
+        def sub_mesh_assert_equal(mesh, exp_in_mesh, exp_out_of_mesh, tensor):
+            if self.rank in mesh:
+                self.assertEqual(tensor, exp_in_mesh)
+            else:
+                self.assertEqual(tensor, exp_out_of_mesh)
+
         mesh = DeviceMesh(self.device_type, [0, 2])
         local_tensor = torch.ones(3, 4)
 
         dtensor = DTensor.from_local(local_tensor, mesh, [Shard(0)])
         self.assertEqual(dtensor.size(), torch.Size([6, 4]))
 
-        if self.rank == 0 or self.rank == 2:
-            self.assertEqual(dtensor.to_local(), torch.ones(3, 4))
-        else:
-            self.assertEqual(dtensor.to_local(), torch.tensor([]))
+        sub_mesh_assert_equal(
+            mesh.mesh,
+            torch.ones(3, 4),
+            torch.tensor([]),
+            dtensor.to_local(),
+        )
 
         # test dtensor created in submesh, the operation should only
         # be applied to the local shard inside the mesh, not the whole
         # world, so only 0/2 really run the computation
-        new_dtensor = dtensor + 2
+        dtensor = dtensor + 2
 
-        if self.rank == 0 or self.rank == 2:
-            self.assertEqual(new_dtensor.to_local(), torch.ones(3, 4) + 2)
-        else:
-            self.assertEqual(new_dtensor.to_local(), torch.tensor([]))
+        sub_mesh_assert_equal(
+            mesh.mesh,
+            torch.ones(3, 4) + 2,
+            torch.tensor([]),
+            dtensor.to_local(),
+        )
+
+        # test scalar return value
+        local_tensor1 = torch.ones(3, 4)
+        local_tensor2 = torch.ones(4, 3)
+        dtensor1 = DTensor.from_local(local_tensor1, mesh, [Shard(0)])
+        dtensor2 = DTensor.from_local(local_tensor2, mesh, [Shard(0)])
+        res = dtensor1.is_same_size(dtensor2)
+        sub_mesh_assert_equal(
+            mesh.mesh,
+            False,
+            True,  # TODO: we might need to communicate the result instead of using the default
+            res,
+        )
 
+        # test 0-d tensor return value
+        dtensor = DTensor.from_local(local_tensor, mesh, [Shard(0)]).sum()
+        sub_mesh_assert_equal(
+            mesh.mesh,
+            torch.tensor(12.),
+            torch.tensor([]),  # TODO: this is actually wrong. should be torch.tensor(0.)
```

Comment:
maybe we need extra handling for scalar tensor? The expected behavior in this case needs to reconsider.